src/adapter/src/catalog.rs

```diff
@@ -2445,6 +2461,25 @@ impl<S: Append> Catalog<S> {
         for (name, value) in system_config {
             catalog.state.insert_system_configuration(&name, &value)?;
         }
+        if !catalog.state.system_config().config_has_synced_once() {
+            if let Some(system_parameter_frontend) = config.system_parameter_frontend {
+                let system_config = catalog.state.system_config();
+                let mut params = SynchronizedParameters::new(system_config.clone());
+
+                if system_parameter_frontend.start_and_initialize().await {
+                    system_parameter_frontend.pull(&mut params);
+
+                    for param in params.modified() {
+                        catalog.state.insert_system_configuration(
+                            param.name.as_str(),
+                            param.value.as_str(),
+                        )?;
+                    }
+
+                    catalog.state.set_config_has_synced_once()?;
```

Comment:
I wasn't able to find a place where this is read and used other than this function. Did I miss something? If not, why do we need to set this?

Reply:
Copy-pasting from the design doc:

> In addition to running that loop, we need to ensure that a one-time sync always runs synchronously as part of the initial `environmentd` creation. In order do that, we will introduce a dedicated  `SystemVar` boolean flag called `config_has_synced_once`. The `mz_environmentd::serve` call will block in cases where this value is `false` until we do the first synchronization. If LD is not reachable at this point in time, we will log an error and fallback to the defaults.